### PR TITLE
Add RES linker

### PR DESCRIPTION
**Pull request by Hazem-IEG**
## Changes proposed in this Pull Request
This PR adds a feature that allows adding custom RES potentials to the network before solving. If the technology is already in the built network, the code will replace its values and if it's not, it will be added. The feature also adds the discount rate as one of the main wildcards in the workflow and thus adding the chosen value to the name of the network.

The config value `custom_data['renewables']` expects a list of technology names to add to the model. If an empty list is passed, no new potentials will be added.

Required:
Adding 2 files per technology per scenario to the folder `resources/custom_renewables` with the name as the following example:
file1: {tech}_{planning_horizons}_{discountrate}_potential.csv format to follow `data/custom_res_potential.csv`
file 2:{tech}_{planning_horizons}_{discountrate}_installable.csv format to follow `data/custom_res_installable.csv`

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Newly introduced dependencies are added to `envs/environment.yaml` and `envs/environment.docs.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`, `config.tutorial.yaml`, and `test/config.test1.yaml`.

### CODE REVIEW

**Comment by energyLS** [comment]
@Hazem-IEG There is one merge conflict, can you have a look at that? After this is solved the CI will run and tell us wether it is fine to merge, and I will be happy to review it :)

**Comment by energyLS** [comment]
> @Hazem-IEG There is one merge conflict, can you have a look at that? After this is solved the CI will run and tell us wether it is fine to merge, and I will be happy to review it :)

I have solved the merge conflict (don't worry: that was done quickly). Now waiting for the CI to pass to review the code

File: Snakefile [hunk 1]
```diff
@@ -365,27 +400,27 @@ rule solve_network:
...
     log:
         solver=RDIR
-        + "/logs/elec_s{simpl}_{clusters}_ec_l{ll}_{opts}_{sopts}_{planning_horizons}_solver.log",
+        + "/logs/elec_s{simpl}_{clusters}_ec_l{ll}_{opts}_{sopts}_{planning_horizons}_{discountrate}_updated_solver.log",
         python=RDIR
-        + "/logs/elec_s{simpl}_{clusters}_ec_l{ll}_{opts}_{sopts}_{planning_horizons}_python.log",
+        + "/logs/elec_s{simpl}_{clusters}_ec_l{ll}_{opts}_{sopts}_{planning_horizons}_{discountrate}_updated_python.log",
         memory=RDIR
-        + "/logs/elec_s{simpl}_{clusters}_ec_l{ll}_{opts}_{sopts}_{planning_horizons}_memory.log",
+        + "/logs/elec_s{simpl}_{clusters}_ec_l{ll}_{opts}_{sopts}_{planning_horizons}_{discountrate}_updated_memory.log",
```
Comment (energyLS) [inline Snakefile:418]: Why is the string "updated" added and is this a permanent solution?

File: config.yaml [hunk 2]
```diff
@@ -5,26 +5,37 @@ results_dir: results/
...
   opts:
   - "Co2L"
   sopts:
-  - "730H"
+  - "3H"
+  
 
 countries: ['MA']
 
+custom_data:
```
Comment (energyLS) [inline config.yaml:29]: Please add all changes in this file also to `https://github.com/pypsa-meets-africa/pypsa-earth-sec/blob/main/test/config.test1.yaml`

Comment (energyLS) [reply config.yaml:29]: Otherwise the CI fails

**Review by energyLS — CHANGES_REQUESTED** [review]
Nice implementation. If the proposed changes are implemented, the CI should work and we can merge it

**Comment by Hazem-IEG** [comment]
CI fixed @energyLS ;)

**Comment by davide-f** [comment]
My feeling is that discountrate is interpreted as a string by snakemake, but we need it as a float in the execution.
I'd make sure to perform the appropriate parsing. I add a review on that in about 1-2 hours

**Comment by Hazem-IEG** [comment]
I finally fixed the error, it is a problem with Biomass that I still didn't figure out yet.  Could you please check the error as you worked more on this carrier? @energyLS

**Comment by energyLS** [comment]
> I finally fixed the error, it is a problem with Biomass that I still didn't figure out yet. Could you please check the error as you worked more on this carrier? @energyLS

Interesting. Yes I will take care

**Comment by Hazem-IEG** [comment]
This should be ready now. Maybe we need to open a new branch for the biomass issue. @energyLS

**Review by energyLS — APPROVED** [review]

**Comment by energyLS** [comment]
> This should be ready now. Maybe we need to open a new branch for the biomass issue. @energyLS

Yes, this PR is now ready. I have opened up a new issue for the biomass issue in #140.
Thanks for the contribution.

**Pull request merged**